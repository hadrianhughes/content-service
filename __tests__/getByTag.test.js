const getByTag = require('../lib/getByTag');

jest.mock('../__mocks__/prismic-javascript.js');

const mockData = require('../__mock-data__/doc.json');
const mockApi = {
  query: q => new Promise ((resolve, reject) => {
    resolve(q);
  })
};

describe('getByTag function', () => {
  it('Should return a promise which resolves with an object when passed an array of 1 item', () => {
    expect(getByTag(mockApi, ['foo']).resolves).toMatchSnapshot();
  });

  it('Should return a promise which resolves with an object when passed an array of multiple items', () => {
    expect(getByTag(mockApi, ['foo', 'bar']).resolves).toMatchSnapshot();
  });

  it('Should return a promise which resolves with an object when passed a type with the tags', () => {
    expect(getByTag(mockApi, ['foo'], 'bar').resolves).toMatchSnapshot();
  });
});
