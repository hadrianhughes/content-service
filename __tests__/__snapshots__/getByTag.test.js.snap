// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getByTag function Should return a promise which resolves with an object when passed a type with the tags 1`] = `undefined`;

exports[`getByTag function Should return a promise which resolves with an object when passed an array of 1 item 1`] = `undefined`;

exports[`getByTag function Should return a promise which resolves with an object when passed an array of multiple items 1`] = `undefined`;
